src/Controllers/AuthUser.php

```diff
@@ -0,0 +1,176 @@
+    {
+        if(!isset($_POST['name'])) {
+            throw new UndefinedControllerException("Trzeba podac imie!");
+        }
+    }
+
+    private static function validateRequestData()
+    {
+        self::validateName($_POST['name']);
+        self::validateEmail($_POST['name']);
+        self::validatePassword($_POST['password'], $_POST['password_confirmation']);
+    }
+
+    /**
+     * @throws UndefinedControllerException
+     */
+    public function signIn(): void
+    {
+        if(empty($_SESSION['userId'])) {
+            View::render('UserManagement.AuthUser.signIn');
+        } else {
+            header('Location: /praktyki24-25/public/index.php?controller=dashboard&do=show');
+            exit();
+        }
+    }
+
+    public function register(): void
+    {
+        self::validateRequestData($_POST);
+        $db = Database::getInstance();
+        $name = $_POST['name'];
+        $email = $_POST['email'];
+        $password = $_POST['password'];
+        $password = password_hash($password, PASSWORD_DEFAULT);
+            
+            $sql = "INSERT INTO users (id, name, email, password) VALUES (null, ?, ?, ?)";
+            $stmt = $db->prepare($sql);
+            $stmt->execute([
+                $name,
+                $email,
+                $password
+            ]);
+//        }
+    }
+
+
+
+    public function login(): void
+    {
+        if (isset($_POST['name']) && isset($_POST['email']) && isset($_POST['password'])) {
+
+            $email = $_POST['email'];
+            $password = $_POST['password'];
+
+            if (true === self::verifyPassword($password, $email)) {
+                $user = User::findByEmail($email);
+                if (isset($user)) {
+                    $userId = $user->getId();
+
+                    if (session_status() === PHP_SESSION_NONE) {
+                        session_start();
+                    }
+                    $_SESSION['userId'] = $userId;
+                    var_dump($_SESSION);
+
+                    header('Location: /praktyki24-25/public/index.php?controller=dashboard&do=show');
+                    exit();
+                }
+            }
+        }
+    }
+
+
+    private static function verifyPassword(mixed $password,mixed $email):bool
+    {
+        $user = User::findByEmail($email);
+        if(isset($user)) {
+            $hash = $user->getPassword();
+        }
+        return password_verify($password, $hash);
+    }
+
+    public function logout(): void
+    {
+        if (session_status() === PHP_SESSION_ACTIVE) {
+            $_SESSION = [];
+            session_destroy();
+
+
+
+            if (ini_get("session.use_cookies")) {
+                $params = session_get_cookie_params();
+                setcookie(session_name(), '', time() - 42000,
+                    $params["path"], $params["domain"],
+                    $params["secure"], $params["httponly"]
+                );
+            }
+        }
+        var_dump($_SESSION);
```

Comment:
Tu jest jakaś pozostałość, jakiś debug. Nie powinno się to tutaj znaleźć - ten var_dump